go/libraries/utils/async/ring_buffer_test.go

```diff
@@ -39,7 +39,7 @@ func TestSingleThread(t *testing.T) {
 			rb := NewRingBuffer(test.allocSize)
 
 			for i := 0; i < test.numItems; i++ {
-				err := rb.Push(i)
+				err := rb.Push(i, rb.epoch)
 				assert.NoError(t, err)
```

Comment:
also `rb.Push(i, rb.epoth+1)` to make sure they are being filtered properly?

Reply:
Added a standalone test for some epoch behavior.